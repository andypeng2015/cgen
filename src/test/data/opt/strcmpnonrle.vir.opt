module foo

export function sw $foo(l %s1, l %s2) {
@2:
  jmp @3(%s2, %s1)
@3(%s2.1, %s1.1):
  %c1.1 = ld.b %s1.1 ; @5
  %c2.1 = ld.b %s2.1 ; @6
  %0 = ne.b %c1.1, 0x0_b ; @16
  br %0, @7, @8
@7:
  %3 = eq.b %c1.1, %c2.1 ; @19
  br %3, @10, @8
@10:
  %5 = add.l %s1.1, 0x1_l ; @21
  %4 = add.l %s2.1, 0x1_l ; @20
  jmp @3(%4, %5)
@8:
  %1 = sub.b %c1.1, %c2.1 ; @17
  %2 = sext.w %1 ; @18
  ret %2
}
