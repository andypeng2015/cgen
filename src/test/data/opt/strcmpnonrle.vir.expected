module foo

export function sw $foo(l %s1, l %s2) {
@13:
  jmp @4(%s2, %s1)
@4(%s2.1, %s1.1):
  %c1.1 = ld.b %s1.1 ; @12
  %c2.1 = ld.b %s2.1 ; @11
  %0 = ne.b %c1.1, 0x0_b ; @14
  br %0, @7, @0
@7:
  %1 = eq.b %c1.1, %c2.1 ; @15
  br %1, @3, @0
@3:
  %2 = add.l %s1.1, 0x1_l ; @16
  %3 = add.l %s2.1, 0x1_l ; @17
  jmp @4(%3, %2)
@0:
  %5 = sub.b %c1.1, %c2.1 ; @19
  %4 = sext.w %5 ; @18
  ret %4
}
